contracts/src/utils/math/storage.rs

```diff
@@ -8,11 +8,13 @@ pub(crate) trait AddAssignUnchecked<T> {
     fn add_assign_unchecked(&mut self, rhs: T);
 }
 
-impl<const B: usize, const L: usize> AddAssignUnchecked<Uint<B, L>>
+impl<T, const B: usize, const L: usize> AddAssignUnchecked<T>
     for StorageUint<B, L>
+where
+    Uint<B, L>: UintTryFrom<T>,
 {
-    fn add_assign_unchecked(&mut self, rhs: Uint<B, L>) {
-        let new_balance = self.get() + rhs;
+    fn add_assign_unchecked(&mut self, rhs: T) {
+        let new_balance = self.get() + Uint::<B, L>::from(rhs);
```

Comment:
This refactoring looks smart.
But initial reason why we are rarely using `Uint::<_,_>::from(..)` method, that it performs conversion in runtime.

In the following example a conversion is done at compile time, with the help of `uint!` macro.

```
self._balances.setter(from).sub_assign_unchecked(uint!(1_U256))
```

Also `Uint::<B, L>::from` has actual conversion code beneath with size checks, and we have `add_unsigned_unchecked` function name

Reply:
This is actually a very good catch!
I'll revert this, and try to think of a way to keep this generic, but won't spend too much time on it.

Reply:
Wait, does this mean that `uint!` is more performant than `U256::from`?

Reply:
Instead of adding `ONE`, we first "create `ONE` from integer checking that integer is not bigger. If it is bigger then panic, if not" add ONE to the value

Reply:
Ofc it won't affect performance dramatically and we wont notice it probably. But I don't see much point of that kind of refactoring. We don't win much in code simplicity.

Reply:
You're right, reverted the change https://github.com/OpenZeppelin/rust-contracts-stylus/pull/467/commits/c3a1599b2397eda8aa71bf29f2f671594c66e7d0